db/fetcher/indexer_iterators.go

```diff
@@ -92,16 +99,62 @@ func (i *eqIndexIterator) Init(ctx context.Context, store datastore.DSReaderWrit
 	return nil
 }
 
-func (i *eqIndexIterator) Next() (core.IndexDataStoreKey, bool, error) {
-	key, hasValue, err := i.queryResultIterator.Next()
-	if hasValue {
+func (i *eqPrefixIndexIterator) Next() indexIterResult {
```

Comment:
suggestion: Same as above. It would be nicer if ther error was kept separate.

Reply:
I followed what seems to be a common way of dealing with iterators. Like `query.Result`.
I think the main advantage of it (besides having shorter returns) it will be easier to use such API with go channels.

Reply:
Do you think we will move to go channels for that API? I'm not sure a channel makes sense here.